File: LeetCode/two-sum.py
# Title:1. Two Sum
# Description:
# Given an array of integers, return indices of the two numbers such that they add up to a specific target.
# You may assume that each input would have exactly one solution, and you may not use the same element twice.
# Example:
# Given nums = [2, 7, 11, 15], target = 9,
# Because nums[0] + nums[1] = 2 + 7 = 9,
# return [0, 1].


# Language: Python
class Solution:
    def twoSum(self, nums, target):
        """
        :type nums: List[int]
        :type target: int
        :rtype: List[int]
        """
        length = len(nums)
        for i in range(0, length):
            for j in range(i + 1, length):
                if nums[i] + nums[j] == target:
                    return [i, j]

solution = Solution()
print(solution.twoSum([2, 7, 11, 15], 9))  

File: LeetCode/go/two-sum/two-sum.go
package twosum

// description: https://leetcode.cn/problems/two-sum
func twoSum(nums []int, target int) []int {
	tempMap := map[int]int{}
	for i := 0; i < len(nums); i++ {
		if targetIndex, ok := tempMap[nums[i]]; ok {
			return []int{i, targetIndex}
		}
		tempMap[target-nums[i]] = i
	}
	return nil
}
